### ISSUE TITLE
handleError not working accordingly to docs?

### ISSUE THREAD

**Issue by sgarbesi** [supercharge/promise-pool#78]
Please see the following example.

When an error is thrown in the `process` Function, it's not being communicated to the `handleError` Function.

From the README.md:

> Providing a custom error handler allows you to exit the promise pool early by throwing inside the error handler function. Throwing errors is in line with Node.js error handling using async/await.

Is this a bug?

Node Version: 18.17.1
Library Version: 3.0.0

**script:**

```javascript
const PromisePool = require("@supercharge/promise-pool").PromisePool;

(async () => {
  await PromisePool.withConcurrency(1)
    .handleError(async (error) => {
      console.log("error", error);
      throw error;
    })
    .for([{ a: 1 }, { b: 2 }, { c: 3 }])
    .process(async (item) => {
      console.log(item);
      throw new Error("test error here");
    });
})();
```

**output:**

```bash
% node test.js
Debugger attached.
{ a: 1 }
{ b: 2 }
{ c: 3 }
Waiting for the debugger to disconnect...
% 
```

**Comment by marcuspoehls** [comment]
@sgarbesi Hey Salvatore, sounds like a bug. I'll double check the test suite for your reproduction and possibly fix the issue. I might be able to get at it next week. I appreciate your help if want to submit a pull request 😊

**Comment by sgarbesi** [comment]
@marcuspoehls Looks like it's the order the chain, this code below works fine it appears.

Hence handleError() comes after for()

```
const PromisePool = require("@supercharge/promise-pool").PromisePool;

(async () => {
  await PromisePool.withConcurrency(1)
    .for([{ a: 1 }, { b: 2 }, { c: 3 }])
    .handleError(async (error) => {
      console.log("error", error);
      throw error;
    })
    .process(async (item) => {
      console.log(item);
      throw new Error("test error here");
    });
})();
```

**Comment by marcuspoehls** [comment]
@sgarbesi good catch! I can confirm that's a big: the order is important (at this point). Calling `.for` after `.handleError` causes the loss of the configured error handler. 

Do you want to submit a pull request for this? 😊

**Comment by marcuspoehls** [comment]
I’ve published version `3.1.0` to fix this order issue. The new version keeps the error handler when calling `pool.handleError` before `pool.for`
